HomeWork1/Taras_Samchuk_HW1/vector.hpp

```diff
@@ -0,0 +1,220 @@
+/*
+          vector.hpp by Sam4uk
+*/
+#pragma once
+#include <cstring>    //+ size_t & memcpy
+#include <iterator>   //+ std::distance
+#include <stdexcept>  //+ std::logic_error
+#include <utility>    //+ std::initializer_list
+
+static constexpr size_t k_grow_factor{2};
+static constexpr size_t k_starting_size{2};
+
+template <class T>
+class Vector {
+ public:
+  // constructor
+  Vector() : _count{0}, _capacity{k_starting_size}, _data{new T[_capacity]} {}
+
+  Vector(const size_t size)
+      : _count{0}, _capacity{size}, _data{new T[_capacity]} {}
+
+  Vector(const std::initializer_list<T>& list)
+      : _count{0}, _capacity{list.size()}, _data{new T[_capacity]} {
```

Comment:
```suggestion
      : Vector(list.size()){
```
старайтесб переиспользовать существующий функционал